chicken-sloths/bangazon-site-sprint3
# Fixes #12: Users can recommend products to one another

# Description
In the `product-details`, authenticated users can recommend products to one another through their email (username). When a user logs in, they will see in the authed user dropdown menu 'Recommendations: [num-of-recs]'. Clicking on that they see all the products that have been recommended to them. They can click on those products to see their details or delete the recommendation.

## Related Ticket(s)
Fixes #12 

## Proposed Changes
A new association between `Product` & `Customer` called `Recommendations` identifies products recommended to particular users. The UI enables making and deleting these recommendations & notifying a user that they have recommendations.

## Expected Behavior
When an authenticated user is logged in & type in the email address of another registered user in the "Recommend to a Friend" input field on any `product-detail` card, that product should appear in that user's `recommendations` view & the `recommendations: [num]` in the user dropdown. If a user is not authenticated, no input fields should be visible. If a user inputs their own email, they can't recommend something to themselves. If a user inputs an email not associated with a registered user, then they can't recommend products to no one.

## Steps to Test Solution

1. `npm run db:gen`
1. `npm start`
1. Choose two users to work with (I used `Nathanial.Nicolas@yahoo.com
` & `Vicenta87@hotmail.com
`) & login in with one of them
1. Select any product
1. Click on `Recommend to a Friend`
1. Type in the email address for the user _you just logged in with_ (e.g., `Nathanial.Nicolas@yahoo.com
`)
1. An error message should appear & no products should have been recommended to that user
1. Type in a random email address
1. An error message should appear
1. Type in the other email you're working with (e.g., `Vicenta87@hotmail.com
`)
1. A success message should appear
1. Log out
1. Log in with the user you just sent that recommendation to (e.g., `Vicenta87@hotmail.com
`)
1. Hover over the user dropdown
1. You should see `recommendations: 1`; click on that
1. The product just recommended should appear
1. Click on delete
1. That product should disappear, a message should appear, & the authed user dropdown should say `recommendations: 0`
1. Refresh the page to confirm that no products reappear!
